File: src/main/java/kr/devport/api/domain/common/cache/CacheNames.java
package kr.devport.api.domain.common.cache;

/**
 * Single source of truth for all cache names used across the application.
 * All @Cacheable, @CacheEvict, and programmatic cache operations must reference these constants.
 */
public final class CacheNames {
    
    // Article domain caches
    public static final String ARTICLES = "articles";
    public static final String TRENDING_TICKER = "trendingTicker";
    
    // Git repository domain caches
    public static final String GIT_REPOS = "gitRepos";
    public static final String TRENDING_GIT_REPOS = "trendingGitRepos";
    public static final String GIT_REPOS_BY_LANGUAGE = "gitReposByLanguage";
    public static final String GITHUB_TRENDING = "githubTrending";
    
    // LLM domain caches
    public static final String LLM_LEADERBOARD = "llmLeaderboard";
    public static final String LLM_BENCHMARKS = "benchmarks";
    public static final String LLM_MODELS = "llmModels";

    // Wiki domain caches
    public static final String WIKI_PROJECTS = "wikiProjects";
    public static final String WIKI_PROJECT_PAGE = "wikiProjectPage";
    
    private CacheNames() {
        throw new UnsupportedOperationException("This is a constants class and cannot be instantiated");
    }
}

File: support/cache/src/main/kotlin/kr/devport/api/domain/common/cache/CacheNames.kt
package kr.devport.api.domain.common.cache

/**
 * Single source of truth for all cache names used across the application. All @Cacheable,
 * @CacheEvict, and programmatic cache operations must reference these constants.
 */
object CacheNames {
    // Article domain caches
    const val ARTICLES = "articles"
    const val TRENDING_TICKER = "trendingTicker"

    // Git repository domain caches
    const val GIT_REPOS = "gitRepos"
    const val TRENDING_GIT_REPOS = "trendingGitRepos"
    const val GIT_REPOS_BY_LANGUAGE = "gitReposByLanguage"
    const val GITHUB_TRENDING = "githubTrending"

    // LLM domain caches
    const val LLM_LEADERBOARD = "llmLeaderboard"
    const val LLM_BENCHMARKS = "benchmarks"
    const val LLM_MODELS = "llmModels"

    // Wiki domain caches
    const val WIKI_PROJECTS = "wikiProjects"
    const val WIKI_PROJECT_PAGE = "wikiProjectPage"
}
